### DISCUSSION TITLE
Possible to integrate justpy (ag-grid) into Dash framework?

### DISCUSSION THREAD

**Discussion by benedictthen1** [elimintz/justpy#429] [Ideas]
Hi, I am currently using the Python Dash framework to build visualizations and would love to integrate justpy (ag-grid) into it. Does anyone know that it is possible? oo

**Comment by elimintz** [comment]
You probably can't run justpy and dash together, but I may be wrong.

**Comment by mmmmaybe** [comment]
Hello @elimintz! Thank you so much for creating justpy! I have no web dev skills or background but justpy made it easy to access to ag-grid and highcharts. 

I was also trying to achieve the same thing as @benedictthen1 because my school project requires to use both dash and ag-grid. I realised that justpy uses **Starlette server** so I tried to mount **flask** and **dash** servers **on Starlette** by adding a few lines of codes in `justpy.py` file. 

Flask works well but for dash, it is somehow showing strange characters instead of showing ploty dash visualizations. 

`justpy.py`
![image](https://user-images.githubusercontent.com/43146701/93000706-b1fa7900-f55c-11ea-978b-ac7f0ae9a0ca.png)
![image](https://user-images.githubusercontent.com/43146701/93000727-c63e7600-f55c-11ea-8fa2-6a1047ee1765.png)

In Browser,
![image](https://user-images.githubusercontent.com/43146701/93000799-3c42dd00-f55d-11ea-9298-e42830c947e4.png)

Does anyone happen to know why? I'm really struggling on this for my school project. T_T

**Comment by elimintz** [comment]
I'm sorry I can't help you with this. Maybe you can convince the prof to use JustPy instead of Dash?😁

**Comment by pabloazurduy** [comment]
> I'm sorry I can't help you with this. Maybe you can convince the prof to use JustPy instead of Dash?😁

Hi @elimintz we where just evaluating both frameworks, I felt a little bit more inclined to use justpy (bc tailwind and also I'm not that familiar with dash, but seems less flexible). Anyway, what are the reasons to choose justpy over dash ? 

Many thanks !!

**Comment by elimintz** [comment]
Hi @pabloazurduy  
Thank you for considering JustPy. 
Just to be clear, there are reasons to prefer dash. One main reason is that it has been around much longer and is more mature. 
That being said, JustPy is more flexible and the syntax is cleaner in my opinion. Also, if you prefer Highcharts as your charting library, then JustPy is a better way to go.
What I would recommend you do is create a small application with both that mimics some of the functionality you are looking for and then decide. Alternatively, run and modify a few examples from both frameworks and see what you like better.
Would be happy to learn what you decide in the end and what were your reasons.
